docs/visual-basic/programming-guide/language-features/data-types/tuples.md

```diff
@@ -0,0 +1,153 @@
+
+All four of these tuples have the same number of fields (referred to as 'arity'), and the types of those fields are identical. Therefore, all of these assignments work:
+
+[!code-vb[Assign](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple3.vb#2)]
+
+Notice that the names of the tuples are not assigned. The values of the fields are assigned following the order of the fields in the tuple.
+
+Tuples of different types or numbers of fields are not assignable:
+
+```vb
+' Does not compile.
+' VB30311: Value of type '(Integer, Integer, Integer)' cannot be converted
+'          to '(Answer As Integer, Message As String)'
+var differentShape = (1, 2, 3)
+named = differentShape
+```
+
+## Tuples as method return values
+
+A method can return only a single value. Frequently, though, you'd like a method call to return multiple values. There are several ways to work around this limitation:
+
+- You can create a custom class or structure whose properties or fields represent values returned by the method. Thus is a heavyweight solution; it requires that you define a custom type whose only purpose is to retrieve values from a method call.
+
+- You can return a single value from the method, and return the remaining values by passing them by reference to the method. This involves the overhead of instantiating a variable and risks inadvertently overwriting the value of the variable that you pass by reference.
+
+- You can use a tuple, which provides a lightweight solution to retrieving multiple return values.
+
+For example, the **TryParse** methods in .NET return a `Boolean` value that indicates whether the parsing operation succeeded. The result of the parsing operation is returned in a variable passed by reference to the method. Normally, a call to the a parsing method such as <xref:System.Int32.TryParse%2A> looks like the following:
+
+[!code-vb[Return](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple-return.vb#1)]
+
+We can return a tuple from the parsing operation if we wrap the call to the <xref:System.Int32.TryParse%2A> method in our own method. In the following example, `NumericLibrary.ParseInteger` calls the <xref:System.Int32.TryParse%2A> method and returns a named tuple with two elements. 
+
+[!code-vb[Return](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple-return.vb#2)]
+
+You can then call the method with code like the following:
+
+[!code-vb[Return](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple-return.vb#2)]
+
+## Visual Basic tuples and tuples in the .NET Framework
```

Comment:
Consider updating the descriptions to compare `System.Tuple` and `System.ValueTuple`. The two types are different. The VB (and C#) teams provided language features that map to System.Tuple.